src/ml/neural_net/image_augmentation.hpp

```diff
@@ -269,7 +269,57 @@ class resize_only_image_augmenter final: public image_augmenter {
   options opts_;
 };
 
+/**
+ * An abstract class that inherits from image_augmenter used to convert
+ * input images, annotations and predictions to shared_float_arrays for
+ * tf_image_augmenter.
+ * Subclass must be written for it if needed. The subclass must implement the
+ * pure virtaul method prepare_augmented_images.
+ */
+class float_array_image_augmenter : public image_augmenter {
+ public:
+  float_array_image_augmenter(const options& opts) : opts_(opts) {}
+
+  const options& get_options() const override { return opts_; }
+
+  result prepare_images(std::vector<labeled_image> source_batch) override;
+
+ protected:
+  /** The output sent from TensorFlow after augmenting the images. */
```

Comment:
This isn't specific to TensorFlow. Let's remove this terminology. Maybe use "the subclass" or "the child class???"